importer/helpers/dagbuilder.go

```diff
@@ -5,6 +5,13 @@ import (
 	"github.com/ipfs/go-ipfs/pin"
 )
 
+// NodeCB is callback function for dag generation
+// the `root` flag signifies whether or not this is
+// the root of a dag.
```

Comment:
i mean this is weird, right, because every node is the root of _a_ dag.

maybe 

```
the `last` flag signifies whether or not this is the last 
(top-most root) node being added. useful for things like
only pinning the first node recursively.
```